src/Microsoft.SqlTools.ServiceLayer/EditData/SmoEditMetadataFactory.cs

```diff
@@ -23,11 +24,21 @@ public class SmoEditMetadataFactory : IEditMetadataFactory
         /// Generates a edit-ready metadata object using SMO
         /// </summary>
         /// <param name="connection">Connection to use for getting metadata</param>
-        /// <param name="objectName">Name of the object to return metadata for</param>
+        /// <param name="objectNamedParts">Split and unwrapped name parts</param>
         /// <param name="objectType">Type of the object to return metadata for</param>
         /// <returns>Metadata about the object requested</returns>
-        public EditTableMetadata GetObjectMetadata(DbConnection connection, string objectName, string objectType)
+        public EditTableMetadata GetObjectMetadata(DbConnection connection, string[] objectNamedParts, string objectType)
         {
+            Validate.IsNotNull(nameof(objectNamedParts), objectNamedParts);
+            if (objectNamedParts.Length <= 0)
+            {
+                throw new ArgumentNullException(nameof(objectNamedParts), SR.EditDataMetadataObjectNameRequired);
```

Comment:
Do we have a better message in SR to provide with this Exception? E.g. "Array cannot be empty"